test/acceptance/tests.py

```diff
@@ -592,6 +623,210 @@ def test_file_upload(self):
+        self.do_training()
+        self.submit_self_assessment(self.SELF_ASSESSMENT)
+
+        return username
+
+    def verify_staff_area_fields(self, username, peer_assessments, submitted_assessments, self_assessment):
+        """
+        Verifies the expected entries in the staff area for peer assessments,
+        submitted assessments, and self assessment.
+
+        Args:
+            username (str): the username of the learner to check
+            peer_assessments: the expected fields in the peer assessment section
+            submitted_assessments: the expected fields in the submitted assessments section
+            self_assessment: the expected fields in the self assessment section
+        """
+        self.staff_area_page.visit()
+        self.staff_area_page.show_learner(username)
+        self.staff_area_page.expand_learner_report_sections()
+        self.assertEqual(peer_assessments, self.staff_area_page.status_text('peer__assessments'))
+        self.assertEqual(submitted_assessments, self.staff_area_page.status_text('submitted__assessments'))
+        self.assertEqual(self_assessment, self.staff_area_page.status_text('self__assessment'))
+
+    def test_staff_override_at_end(self):
+        """
+        Scenario: complete workflow with staff override at the very end
+
+        Given that I have created a submission, completed training, and done a self assessment
+        And a second learner has also created a submission, training, and self assessment
+        Then I can assess a learner
+        And when another learner can assess me
+        Then I see my score based on the peer assessment
+        And when a staff member overrides the score
+        Then I see the staff override score
+        And all fields in the staff area tool are correct
+        """
+        # Create a learner with submission, training, and self assessment completed.
+        learner = self.do_submission_training_self_assessment(self.LEARNER_EMAIL, self.LEARNER_PASSWORD)
+
+        # Now create a second learner so that learner 1 has someone to assess.
+        # The second learner does all the steps as well (submission, training, self assessment, peer assessment).
+        self.do_submission_training_self_assessment("learner2@foo.com", None)
+        self.do_peer_assessment(options=self.PEER_ASSESSMENT)
+
+        # Go back to the first learner to complete her workflow.
+        self.login_user(learner)
+
+        # Learner 1 does peer assessment of learner 2 to complete workflow.
+        self.do_peer_assessment(options=self.SUBMITTED_ASSESSMENT)
+
+        # If learner does not now have a score, it means that "extra" submissions are in the system,
+        # and more need to be scored. Create additional learners and have them grade until learner has
+        # a grade (stopping after 5 additional learners).
+        count = 0
+        while self.grade_page.wait_for_page().score is None and count < 6:
```

Comment:
It would be great if this logic was in a helper method so it can be reused, and so that it doesn't complicate this test.